src/main/java/com/evanlennick/retry4j/AsyncCallExecutor.java

```diff
@@ -75,31 +103,60 @@ private void executeFuture(Callable<T> callable, String callName, CallExecutor<T
         }
     }
 
+    @Deprecated
     public AsyncCallExecutor<T> afterFailedTry(RetryListener<T> listener) {
         this.afterFailedTryListener = listener;
         return this;
     }
 
+    @Deprecated
     public AsyncCallExecutor<T> beforeNextTry(RetryListener<T> listener) {
         this.beforeNextTryListener = listener;
         return this;
     }
 
+    @Deprecated
     public AsyncCallExecutor<T> onCompletion(RetryListener<T> listener) {
         this.onCompletionListener = listener;
         return this;
     }
 
+    @Deprecated
```

Comment:
honestly i think id be fine with just removing these entirely instead of deprecating... this library is still not 1.x.x and the API is unstable, id rather just go straight to the ideal state than worry about not breaking backwards compatibility at this point